src/AspNet.Security.OAuth.Apple/Internal/DefaultAppleClientSecretGenerator.cs

```diff
@@ -86,7 +104,8 @@ public override async Task<string> GenerateAsync([NotNull] AppleGenerateClientSe
             {
                 tokenDescriptor.SigningCredentials = CreateSigningCredentials(context.Options.KeyId!, algorithm);
 
-                clientSecret = _tokenHandler.CreateEncodedJwt(tokenDescriptor);
+                var tokenHandler = context.Options.JwtSecurityTokenHandler ?? new JwtSecurityTokenHandler();
```

Comment:
I wouldn't care much about this option being null. We can still enforce it at validation time if we want to be 100% sure.

Reply:
As above, taking this out locally breaks some tests because the validation code doesn't always run unless it's a full in-the-HTTP-pipeline end-to-end test.